extensions/ql-vscode/src/databases-ui.ts

```diff
@@ -213,113 +330,174 @@ export class DatabaseUI extends DisposableObject {
       showAndLogErrorMessage(e.message);
       return undefined;
     }
-  }
-
-  handleChooseDatabaseInternet = async (): Promise<DatabaseItem | undefined> => {
-    return await promptImportInternetDatabase(this.databaseManager, this.storagePath);
-  }
+  };
+
+  handleChooseDatabaseInternet = async (): Promise<
+    DatabaseItem | undefined
+  > => {
+    return await promptImportInternetDatabase(
+      this.databaseManager,
+      this.storagePath
+    );
+  };
 
   handleChooseDatabaseLgtm = async (): Promise<DatabaseItem | undefined> => {
-    return await promptImportLgtmDatabase(this.databaseManager, this.storagePath);
-  }
+    return await promptImportLgtmDatabase(
+      this.databaseManager,
+      this.storagePath
+    );
+  };
 
   private handleSortByName = async () => {
     if (this.treeDataProvider.sortOrder === SortOrder.NameAsc) {
       this.treeDataProvider.sortOrder = SortOrder.NameDesc;
     } else {
       this.treeDataProvider.sortOrder = SortOrder.NameAsc;
     }
-  }
+  };
 
   private handleSortByDateAdded = async () => {
     if (this.treeDataProvider.sortOrder === SortOrder.DateAddedAsc) {
       this.treeDataProvider.sortOrder = SortOrder.DateAddedDesc;
     } else {
       this.treeDataProvider.sortOrder = SortOrder.DateAddedAsc;
     }
-  }
+  };
 
   private handleUpgradeCurrentDatabase = async (): Promise<void> => {
-    await this.handleUpgradeDatabase(this.databaseManager.currentDatabaseItem);
-  }
-
-  private handleUpgradeDatabase = async (databaseItem: DatabaseItem | undefined): Promise<void> => {
-    if (this.queryServer === undefined) {
-      logger.log('Received request to upgrade database, but there is no running query server.');
-      return;
-    }
-    if (databaseItem === undefined) {
-      logger.log('Received request to upgrade database, but no database was provided.');
-      return;
-    }
-    if (databaseItem.contents === undefined) {
-      logger.log('Received request to upgrade database, but database contents could not be found.');
-      return;
-    }
-    if (databaseItem.contents.dbSchemeUri === undefined) {
-      logger.log('Received request to upgrade database, but database has no schema.');
-      return;
-    }
+    await this.handleUpgradeDatabase(
+      this.databaseManager.currentDatabaseItem,
+      []
```

Comment:
For consistency with how multi-query commands were handled, shouldn't this empty list be either `undefined` or a singleton list of `this.databaseManager.currentDatabaseItem`?

Reply:
I guess based on the way it's implemented it doesn't really matter. I'm very much not a fan of this pattern of `(singleThing: T, multipleThings: T[])` where allegedly one or the other can be undefined, and usually the `multipleThings` includes the `singleThing`, but vscode's API imposes it on us at the command callback interface, so I guess it's fine.

Reply:
I'm disappointed with how under-specified this part of the API is. There's no clear location where it's described how the multiselection passes arguments as far as I can tell. I had to figure this out trial and error.